src/DataObject/DataObjectDocument.php

```diff
@@ -144,7 +144,18 @@ public function shouldIndex(): bool
 
         // If an anonymous user can't view it
         $isPublic = Member::actAs(null, function () use ($dataObject) {
-            return $dataObject->canView();
+            // Need to make sure that the version of the DataObject that we access is always the LIVE version
+            return Versioned::withVersionedMode(function () use ($dataObject): bool {
```

Comment:
```suggestion
            return Versioned::withVersionedMode(static function () use ($dataObject): bool {
```

Feels like this could be `static` unless I'm missing something?